frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.tsx

```diff
@@ -34,9 +36,20 @@ interface DispatchProps {
 
 type Props = MainNavbarProps & StateProps & DispatchProps;
 
-function mapStateToProps(state: State) {
+function getModelDetailPageCard(state: State, params: { slug?: string }) {
+  const entityId = Urls.extractEntityId(params.slug);
+  return Questions.selectors.getObject(state, { entityId });
+}
+
+function mapStateToProps(
+  state: State,
+  { location, params }: MainNavbarOwnProps,
+) {
+  const isModelDetailPage = isModelDetailPathname(location.pathname);
   return {
-    question: getQuestion(state),
```

Comment:
I think we should use `Question` where possible, e.g. `new Question(card, getMetadata(state))`

Reply:
@ranquild we only need `id` and `collection_id` here; I'm not sure if it's worth making the navbar depend on `metadata` here. WDYT?

Reply:
I think less "raw" cards we have the better... but I'm okay either way